addon/services/unified-event-handler.js

```diff
@@ -145,6 +147,21 @@ export default Ember.Service.extend(Ember.Evented, {
     this.on(domHandlerInfo.emberEventName, callback);
   },
 
+  willDestroy() {
+    let handlerMap = this[_HANDLER_MAP];
+
+    for (let target in handlerMap) {
+      let handlerTarget = handlerMap[target];
+      for (let eventName in handlerTarget) {
+        let handlerInfo = handlerTarget[eventName];
+        let emberHandlers = handlerInfo.emberHandlers.slice(0);
```

Comment:
Clone array since unregister mutates handlerInfo.emberHandlers as we enumerate it.